src/analysisd/eventinfo.c

```diff
@@ -906,7 +906,7 @@ char* ParseRuleComment(Eventinfo *lf) {
         if (n + (z = strlen(str)) >= OS_COMMENT_MAX)
             return strdup(lf->generated_rule->comment);
 
-        strncpy(&final[n], str, z);
+        strncat(final, str, OS_COMMENT_MAX - n);
```

Comment:
Not sure I understood this: isn't this supposed to copy `str` to `final + n`? I.e. not the beginning of `final`.

Reply:
Hi @RamiroRD 

The behavior of L909 is the same, the main difference is that `strncat` will guarantee us that the final string always has the null-terminated string, and, in my opinion, it is clearer than doing the `strncat` implementation yourself. 

Best regards
Hanes